### DISCUSSION TITLE
Improving syncing using websockets.

### DISCUSSION THREAD

**Discussion by peshala-prabhapoorna** [super-productivity/super-productivity#3546] [Ideas]
Hi everyone, 
I've been using the app on Linux and Android for 3 days now. Without a doubt this is the best task management application I've come across for my use case. Massive thank you to all the contributors. 

I use Dropbox to sync data between my laptop and android phone. During the short time I've been using the application a non trivial number of conflicts between local and remote data have occurred. I assume that this is due to syncing being performed using **http**. 

I was wondering whether a websocket based syncing option could be implemented. I think this would make the experience much smoother. I understand that this would require the admins to deploy a server and that it comes with costs. Also some users might have privacy concerns about sending data to a remote server. But I think it would be better to have it available as an option to users who want the smoothness of syncing. And as for costs a donation/sponsor based system or a very small monthly/yearly subscription are the only options I guess. 

I'm just blabbering and thinking out loud here. Feel free to correct me if my assumptions are wrong and any sort of criticism is welcome.

**Comment by johannesjo** [comment]
Thanks for the interesting suggestion! There are plans for something like this, but not sure when I will get to this :)

**Reply by peshala-prabhapoorna** [reply]
I would like to try to implement this. Would you mind if I start a different project using Super Productivity as a foundation? I'm not familiar with the etiquette of going about doing these sort of projects. So, I'm open to suggestions/comments/guidance.

**Reply by johannesjo** [reply]
No, I wouldn't mind. Since this is Open Source you can do anything you like. But thank you very much for asking! :)
